saleor/checkout/utils.py

```diff
@@ -705,3 +706,7 @@ def validate_variants_in_checkout_lines(lines: Iterable["CheckoutLineInfo"]):
                 )
             }
         )
+
+
+def net_or_gross(taxed_money: "TaxedMoney", taxes_included: bool) -> Money:
```

Comment:
I am not sure that this should be a part of  this feature as we're fixing incorrect assumption for voucher inside a feature for new way of calculating a taxes. 
This needs to be tested very hard, as it changes the current way of working of vouchers. I would propose, to separate these changes and create a new PR for them.

Reply:
Also, I would change the name of function, to something without `or`. Something like get_base_for_discount or something. Then you could have a flag `from_gross` to determine if it should return .gross value or .net value